sycl/source/detail/event_impl.cpp

```diff
@@ -375,6 +375,12 @@ pi_native_handle event_impl::getNative() {
     MIsInitialized = true;
     auto TempContext = MContext.get()->getHandleRef();
     Plugin.call<PiApiKind::piEventCreate>(TempContext, &MEvent);
+    // See an assert in sycl::detail::getImplBackend.
+    assert(!MHostEvent && "Can't get native event from a host event!");
+    assert(!isDiscarded() && "Can't ask getNative of a discarded event!");
+    // Wouldn't be true if MHostEvent could be true.
+    assert(MState == HES_Complete && "Expected to have a completed event!");
+    Plugin.call<PiApiKind::piEventSetStatus>(MEvent, PI_EVENT_COMPLETE);
```

Comment:
Can we instead change the "piEventCreate" to create a completed event?
BTW, in what circumstances is it called? How is such event used afterwards?

Reply:
> BTW, in what circumstances is it called? How is such event used afterwards?

Not sure, I've only got a reduced example from the customer. I'd expect (without any grounds for those expectations) something like this is possible

```
  auto event = [&]() {
    if (needs_some_preprocessing) {
      return q.submit();
    } else {
      return event();
    }
  } ();
  
  q.parallel_for<MAIN>(..., event);
```

> Can we instead change the "piEventCreate" to create a completed event?

From what I see in the codebase currently, that's possible although I don't see any substantial benefits.

Reply:
I mean why do we even need to create a native event (call to piEventCreate) when we need to create a completed event, for which I assume there is no SYCL API to reset such event? We'd just mark the SYCL event as completed without going to native plugins.

>> Can we instead change the "piEventCreate" to create a completed event?

> From what I see in the codebase currently, that's possible although I don't see any substantial benefits.

To continue deprecating and removing the piEventSetStatus

Reply:
> I mean why do we even need to create a native event (call to piEventCreate) when we need to create a completed event, for which I assume there is no SYCL API to reset such event? We'd just mark the SYCL event as completed without going to native plugins.

Because SYCL spec says that such event 

> is constructed as though it was created from a default-constructed queue. Therefore, its backend is the same as the backend from the default device.

which means it is subject to interop API and its state can be modified using native backend APIs (which seems to be what the customer is doing).

Reply:
OK, makes sense. So what about changing piEventCreate instead?

Reply:
> So what about changing piEventCreate instead?

I'm already working on that.